### DISCUSSION TITLE
How to add device (isMobile) specific code to an SSG page in NextJS?

### DISCUSSION THREAD

**Discussion by cbdeveloper** [vercel/next.js#40381] [Help] [answered]
I have a NextJS app page that should be SSG. I'll fetch all the data that it needs during build time.

But a small part of the code (social media share link) needs information about the device of the user. Basically, I want to know whether I'm on desktop or mobile.

What is the recommended approach for that scenario?

Should I render one of the two possibilities on build time and add some client `useEffect` code to detect and change it if necessary?

**Comment by icyJoseph** [comment, marked as answer]
- useEffect, if the data you need can only be gained with JavaScript
- CSS, if its purely visual, then you can of course hide/show things with media queries and what not

It is important to make sure that the client's first rendered frame is the same as that which the server sends otherwise you'll get HTML mismatched errors.
